codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -148,6 +149,7 @@
     ByteBuf cumulation;
     private Cumulator cumulator = MERGE_CUMULATOR;
     private boolean singleDecode;
+    private boolean decodeWasNull;
```

Comment:
I think this part was not merged properly. This field was intentionally removed in commit made since the file was originally copied.

Reply:
@njhill this variable controls reduce ctx.read() method execution when calling channelReadComplete method.

Reply:
@SteNicholas maybe I don't follow your change completely but it seems that you didn't properly incorporate the optimization done in 6b6475fb565551c36f372fe37683d98df6674161 when merging. `decodeWasNull` was replaced by `firedChannelRead` and it's not clear to me why this shouldn't remain.